src/main/scala/mesosphere/marathon/state/Reservation.scala

```diff
@@ -0,0 +1,82 @@
+package mesosphere.marathon
+package state
+
+import mesosphere.marathon.core.instance.{LocalVolumeId, Instance => CoreInstance, Reservation => CoreReservation}
+import mesosphere.marathon.core.task.Task
+import play.api.libs.json._
+
+/**
+  * Storage model for [[CoreReservation]] with an optional id since older Marathon versions did not
+  * persist reservation ids.
+  */
+case class Reservation(volumeIds: Seq[LocalVolumeId], state: CoreReservation.State, id: Option[CoreReservation.Id]) {
```

Comment:
Can you help me understand why id is optional? It seems like since this is a new class, we can store it this way during the initial migration. Am I missing something?

Reply:
Sent an invite to chat about this your time tomorrow afternoon

Reply:
I started with a migration and a non optional id but found that it broke `Migration17`, `Migration18` and `Migration18100`. Then I figured that there is no need for a migration since we can always infer the reservation id as we already do on master. So I introduced the `state.Reservation` in line with the `state.Instance`. These are the storage models. `core.Reservation` and `core.Instance` are the models we use internally. `core.Reservation` as a non-optional id. It *has* to be set. When we load a reservation and no id was saved we set it.

This means that all reservations will have a persisted id after the first upgrade the has this patch. Once migrations < 1.9 are dropped we can also drop the optional id in `state.Reservation`.